app/controllers/repository_rows_controller.rb

```diff
@@ -51,6 +51,27 @@ def create
                 repository_column: column
               }
             )
+          elsif column.data_type == 'RepositoryAssetValue'
+            asset = Asset.new(file: value,
+                              created_by: current_user,
+                              last_modified_by: current_user,
+                              team: current_team)
+            if asset.save
+              asset.post_process_file(current_team)
+            else
+              errors[:repository_cells] << {
+                "#{column.id}": { data: asset.errors.messages[:file].first }
+              }
+            end
+            cell_value = RepositoryAssetValue.new(
```

Comment:
For this one and the others cell_value stuff we can extract in a method that creates a new object of whatever type. We duplicate this in a lot of places. I wonder if repository object design pattern would help us here.